src/qt/emuwindow.cpp

```diff
@@ -94,3 +98,44 @@ void EmuWindow::closeEvent(QCloseEvent *event)
     event->accept();
     is_running = false;
 }
+
+double EmuWindow::get_frame_rate()
+{
+    //Returns the framerate since old_frametime was set to the current time
+    chrono::system_clock::time_point now = chrono::system_clock::now();
+    chrono::duration<double> elapsed_seconds = now - old_frametime;
+    return (1 / elapsed_seconds.count());
+}
+
+void EmuWindow::update_window_title()
+{
+    /*
+    Updates window title every second
+    Framerate displayed is the average framerate over 1 second
+    */
+    chrono::system_clock::time_point now = chrono::system_clock::now();
+    chrono::duration<double> elapsed_update_seconds = now - old_update_time;
+
+    double framerate = get_frame_rate();
+    framerate_avg = (framerate_avg + framerate) / 2;
+    if (elapsed_update_seconds.count() >= 1.0)
+    {
+        int rf = (int) round(framerate_avg); // rounded framerate
+        string new_title = "DobieStation - FPS: " + to_string(rf);
+        setWindowTitle(QString::fromStdString(new_title));
+        old_update_time = chrono::system_clock::now();
+        framerate_avg = framerate;
+    }
+}
+
+void EmuWindow::limit_frame_rate()
+{
+    while (get_frame_rate() > 60.0)
+    {
```

Comment:
For control-flow, curly braces should only be used for multiple-line blocks. Another stylistic change that needs to be added.
Requested change: ```while (get_frame_rate() > 60.0);```

(While not necessary to worry about for now, it's worth noting that this will need to be revised for when multithreading is implemented. It will be better to make the emulation thread sleep rather than enter a busy loop.)